sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -962,6 +962,11 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
         IsNotNull(e)
       case SqlBaseParser.NULL =>
         IsNull(e)
+      case SqlBaseParser.DISTINCT =>
+        ctx.NOT match {
+          case null => Not(EqualTo(e, expression(ctx.right)))
```

Comment:
BTW, Could you please add a comment in [line 931](https://github.com/ptkool/spark/blob/faf5ebe4f3fad74e6b44de63c8314a853b895e55/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L937) regarding this new syntax support?

Nit: could you please follow how we deal with `IS NOT NULL`?
```Scala
  case SqlBaseParser.DISTINCT if ctx.NOT != null =>
    EqualNullSafe(e, expression(ctx.right))
  case SqlBaseParser.DISTINCT =>
    Not(EqualNullSafe(e, expression(ctx.right)))
```